modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/SourceContext.java

```diff
@@ -168,8 +169,12 @@ public static void updateState(NHttpConnection conn, ProtocolState state) {
             if(sourceContext.getState() == state) {
                 return;
             }
-            if (sourceContext.getSourceConfiguration().isCorrelationLoggingEnabled()) {
+            if (PassThroughCorrelationConfigDataHolder.isEnable()) {
                 long lastStateUpdateTime = sourceContext.getLastStateUpdatedTime();
+                if (PassThroughCorrelationConfigDataHolder.isToggled()) {
```

Comment:
let's add a comment here why we are doing this here.

Reply:
Sure.